### PR TITLE
Added my quicksort implementation

### PR DESCRIPTION
**Pull request by Glazomer**

### CODE REVIEW

File: sorts/quicksort.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+module.exports = function QuickSort(arr: number[]): number[] {
+  const stack: number[][] = [];
+  if (arr.length > 1) stack.push([0, arr.length - 1]);
+  while (stack.length) {
+    // left and right boundaries of unsorted subsequences
+    const [left, right] = stack.pop();
+    let l = left,
+      r = right,
+      m = Math.floor((l + r) / 2);
```
Comment (vitkarpov) [inline sorts/quicksort.ts:22]: I'd call it "pivot" not "middle".

File: sorts/quicksort.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    --r; // r or right most boundary reserved for our "middle" or pivot element, that will be swapped into correct place
+
+    while (l <= r) {
+      if (arr[l] >= arr[right]) {
+        [arr[r], arr[l]] = [arr[l], arr[r]];
+        --r;
+      } else {
+        ++l;
+      }
+    }
```
Comment (vitkarpov) [inline sorts/quicksort.ts:39]: It's better to strip this out into `partition` function. First of all, it'd be easier to read, and it's a partition phase anyway.

**Review by vitkarpov — COMMENTED** [review]
IMHO, that would be much better with recursion & partition function.

File: sorts/quicksort.ts (hunk 1 above)
Comment (Glazomer) [reply sorts/quicksort.ts:22]: Good Idea

File: sorts/quicksort.ts (hunk 2 above)
Comment (Glazomer) [reply sorts/quicksort.ts:39]: How exactly you suggest to do that? It's just one while loop and one if-else statement. Or wrap the whole part of algorithm?

Comment (Glazomer) [reply sorts/quicksort.ts:39]: I think using recursion is unnecessary here

File: sorts/quicksort.ts (hunk 1 above)
Comment (Glazomer) [reply sorts/quicksort.ts:22]: No, this should be called middle, because we will determine the pivot later in algorithm

**Pull request merged**